src/command_modules/azure-cli-botservice/azure/cli/command_modules/botservice/_params.py

```diff
@@ -0,0 +1,130 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.arguments import CLIArgumentType
+
+from azure.cli.core.commands.parameters import (
+    resource_group_name_type,
+    get_enum_type,
+    get_three_state_flag)
+
+name_arg_type = CLIArgumentType(metavar='NAME', configured_default='botname', id_part='Name')
+
+
+# pylint: disable=line-too-long,too-many-statements
+def load_arguments(self, _):
+    with self.argument_context('bot') as c:
+        c.argument('resource_group_name', arg_type=resource_group_name_type)
+        c.argument('resource_name', options_list=['--name', '-n'], help='the Resource Name of the bot.', arg_type=name_arg_type)
+
+    with self.argument_context('bot create') as c:
+        c.argument('sku_name', options_list=['--sku'], arg_type=get_enum_type(['F0', 'S1']), help='the Sku of the Bot', arg_group='Registration Bot Specific')
+        c.argument('kind', options_list=['--kind', '-k'], arg_type=get_enum_type(['registration', 'function', 'webapp']), help='The Kind of the Bot.')
+        c.argument('display_name', help='the Display Name of the bot.If not specified, defaults to the name of the bot.', arg_group='Registration Bot Specific')
+        c.argument('description', options_list=['--description', '-d'], help='the Description of the bot.', arg_group='Registration Bot Specific')
+        c.argument('endpoint', options_list=['-e', '--endpoint'], help='the Messaging Endpoint of the bot.', arg_group='Registration Bot Specific')
+        c.argument('msa_app_id', options_list=['--appid'], help='the msa account id to be used with the bot.')
+        c.argument('password', options_list=['-p', '--password'], help='the msa password for the bot from developer portal.')
+        c.argument('storageAccountName', options_list=['-s', '--storage'], help='Storage Account Name to be used with the bot.If one is not provided, a new account will be created.', arg_group='Web/Function Bot Specific')
+        c.argument('tags', help='set of tags to add to the bot.')
+        c.argument('language', help='The language to be used to create the bot.', options_list=['--lang'], arg_type=get_enum_type(['Csharp', 'Node']), arg_group='Web/Function Bot Specific')
+        c.argument('appInsightsLocation', help='The location for the app insights to be used with the bot.', options_list=['--insights'], arg_group='Web/Function Bot Specific')
```

Comment:
Is this a location (i.e. westus) or an App Insights ID?

Reply:
it's the location where the app insights account will be provisioned. so basically westus etc. i will change it to --insights-location to make that clearer in the parameter

Reply:
In that case, you should look at the definition for `get_location_type` in `azure.cli.core.parameters` and add the completer and type annotations specified there.

Reply:
i am already using get_location_type for the main botservice location. the problem here is that we provision an application insights account as well with a botservice and appinsights is not available for all regions. They also dont have a cli/python sdk for management so i can't query them to get avaialable locations. For now, i have added an enum with the avaialable locations. 
does that work ? (alternatively, i can also add a command type similar to get_location_type which uses a pre-defined list of locations for the completer)

Reply:
That sounds fine.